core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -254,6 +255,7 @@ public boolean isNameEditable() {
      */
     @RequirePOST
     @Restricted(NoExternalUse.class)
```

Comment:
Note that these restrictions become easily bypassable via casting to `Renamable` with the new changes. Perhaps the new interface should be marked as `@Restricted(ProtectedExternally.class)`? Even then there would be some issues though, see https://github.com/jenkinsci/lib-access-modifier/issues/22 and https://github.com/jenkinsci/lib-access-modifier/issues/21.

Reply:
added `@Restricted(ProtectedExternally.class)` in b7eed6f13284dc3ab849cd10b23aad1dadc3c7f1